pkg/api/http/http.go

```diff
@@ -99,6 +99,7 @@ const (
 	workflowComponent        = "workflowComponent"
 	workflowName             = "workflowName"
 	instanceID               = "instanceID"
+	nonRecursive             = "non_recursive"
```

Comment:
Can we make this `recursive` instead of `non_recursive`? It's harder for people to understand negatives.

Reply:
I kind of incline towards `recursive` set to `true` by default. Understanding this was quite easier for me.
Is there any reasoning behind keeping all flags to false by default, apart from general practice?

Reply:
I personally agree with Mukundan that it's more intuitive when the default value of a boolean is false.

If the default value is "true", then you are introducing 3 states: set true, set false, null.